### ISSUE TITLE
Script Migrations don't support placeholders

### ISSUE THREAD

**Issue by dvdtknsn** [flyway/flyway#3081]
Using Flyway 7.0.0 - command line

I'm running sqlcompare.exe as a .bat script migration. This allows me to deploy using a combination of migrations, callbacks and a state-based methodology.

The nature of a state-base deployment is that it requires a comparison between the source and target to determine what the changes are, and if there are any changes a SQL script is generated and run to apply the changes.

Therefore this has to be a repeatable migration. And one that is run every time as the checksum of the batch file itself is irrelevant to the changes that have been applied.

I therefore wrote this script R__SQLCompare.bat
```
REM We use the ${flyway:timestamp} trick to ensure the the migration script checksum changes each time and forces the migration to run every time

echo placeholders don't seem to work as expected: target:${flyway:database} >> log.txt
REM "C:\Program Files (x86)\Red Gate\SQL Compare 14\sqlcompare.exe" /s1:. /db1:FlywayDB /s2:. /db2:${flyway:database} /scriptFile=DeployScript.sql /sync /out:outputlog.txt /f

REM So I am hard coding for now, but I would imagine that we'd want to be able to pass in source and target parameters.
"C:\Program Files (x86)\Red Gate\SQL Compare 14\sqlcompare.exe" /s1:. /db1:FlywayDB /s2:. /db2:FlywayDBProd /scriptFile=DeployScript.sql /sync /out:outputlog.txt /f
echo Errorlevel is: %ERRORLEVEL%

REM SQL Compare uses 63 as an exit code when differences are found, so we detect this and set to 0 so not to fail the migration
IF %ERRORLEVEL% EQU 63 (
   SET ERRORLEVEL=0
)
```

 ${flyway:timestamp} was used to ensure that the repeatable migration is run each time. This worked, although I've been told that this only works by accident (but please don't change this!).

Issues:
1) SQLCompare's console output wasn't displayed, which made it hard to troubleshoot. Sqlcompare can output its log to a file so there was a workaround. the Flyway -X flag also worked, but for me the basic info that SQLCompare outputs isn't debug info - it's just standard informational output that should be visible to me without -X

2) I was unable to pass in placeholders to my sqlcompare command line invocation. SQL Compare requires a source and target to be specified. The source will eventually be a scripts folder in my packaged Flyway project, so that will end up being a relative path, but the target database connection details will need to be passed in. I expected ${flyway:database} to work, but it didn't. Also, I would have liked there to be a ${flyway:serverInstance} too, although I know that I can create and pass in my own custom placeholders.

In summary, this feature appears very close to being a viable vehicle with which to deliver a state-based deployment capability in the context of Flyway deployments.

**Comment by DoodleBobBuffPants** [comment]
For now we have documented the behavior of placeholders in script migrations [here](https://flywaydb.org/documentation/concepts/migrations#important-notes-1)

**Comment by Phil-Factor** [comment]
+1 for this.   At the moment all communication with scripts has to be done via files, which is awkward and makes debug difficult. All my DOS and Powershell routines will need parameters to be passed to them via PlaceHolders, just as happens with my SQL Scripts. I would expect that Dos's ECHO and application output should be passed on by Flyway without having to use the -X flag.  Same with PowerShell. This will allow scripts to pass information back and forth.
